src/shared/theme/__tests__/validate.test.ts

```diff
@@ -13,51 +13,76 @@ afterAll(() => {
   });
-});
 
-test('validateOverride prints warning on unrecognized token and drops it', () => {
-  const validated = validateOverride(override, ['background'], ['navigation']);
+  test('prints warning on unrecognized token and drops it', () => {
+    const validated = validateOverride(override, ['background'], ['navigation']);
 
-  expect(spy).toBeCalled();
-  expect(Object.keys(validated.tokens)).toHaveLength(0);
-  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-  expect(Object.keys(validated.contexts!.navigation!.tokens!)).toHaveLength(0);
-});
+    expect(spy).toBeCalled();
+    expect(Object.keys(validated.tokens)).toHaveLength(0);
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    expect(Object.keys(validated.contexts!.navigation!.tokens!)).toHaveLength(0);
+  });
 
-test('validateOverride prints warning on unrecognized context ID and drops it', () => {
-  const validated = validateOverride(override, Object.keys(override.tokens), ['a different context ID']);
+  test('prints warning on unrecognized context ID and drops it', () => {
+    const validated = validateOverride(override, Object.keys(override.tokens), ['a different context ID']);
 
-  expect(spy).toBeCalled();
-  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-  expect(Object.keys(validated.contexts!)).toHaveLength(0);
+    expect(spy).toBeCalled();
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    expect(Object.keys(validated.contexts!)).toHaveLength(0);
+  });
+
+  test('throws error for missing or wrong tokens field', () => {
+    expect(() => validateOverride({} as unknown as Override, [], [])).toThrow(
+      'Missing required "tokens" object field in {}'
+    );
+    expect(() => validateOverride({ tokens: [] } as unknown as Override, [], [])).toThrow(
+      'Missing required "tokens" object field in {"tokens":[]}'
+    );
+  });
 });
 
-test('throws error for missing or wrong tokens field', () => {
-  expect(() => validateOverride({} as unknown as Override, [], [])).toThrow(
-    'Missing required "tokens" object field in {}'
-  );
-  expect(() => validateOverride({ tokens: [] } as unknown as Override, [], [])).toThrow(
-    'Missing required "tokens" object field in {"tokens":[]}'
-  );
+describe('getThemeFromPreset', () => {
+  test('returns default theme if themeId is not specified', () => {
+    const theme = getThemeFromPreset(presetWithSecondaryTheme);
+    expect(theme).toEqual(presetWithSecondaryTheme.theme);
+  });
+
+  test('returns default theme if themeId matches the default one', () => {
+    const theme = getThemeFromPreset(presetWithSecondaryTheme, 'root');
+    expect(theme).toEqual(presetWithSecondaryTheme.theme);
+  });
+
+  test('returns secondary theme if themeId matches the a secondary theme', () => {
+    const theme = getThemeFromPreset(presetWithSecondaryTheme, 'secondary');
+    expect(theme).toEqual(presetWithSecondaryTheme.secondary?.[0]);
```

Comment:
I'd recommend to replace the question mark with an exclamation mark here, to avoid a potential false positive test result if the value is `undefined`.

Reply:
Addressed